packages/export/src/nodes/pivot.js

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+import ContextureClient from 'contexture-client'
+import { andGroup, runWith, setFilterOnly } from '../utils.js'
+
+// TODO import from contexture-client
+export let getKey = (x) => x.keyAsString || x.key
+
+// TODO import from contexture-client
+// Similar to Tree.lookup but path is a drilldown which uses keyAsString or key
+let resultsForDrilldown = (type, drilldown, results) => {
+  if (_.isEmpty(drilldown) || !results) return results
+
+  let key = _.first(drilldown)
+  let groups = _.get(type, results)
+  let match = _.find((node) => getKey(node) === key, groups)
+
+  return resultsForDrilldown(type, drilldown.slice(1), match)
+}
+
+export default async ({ service, tree, exportAllPages, ...node }) => {
+  let run = (node) => runWith(service, tree, node)
+
+  let getContextureClient = (tree, node) =>
+    ContextureClient(
+      {
+        service,
+        debounce: 0,
+      },
+      andGroup(setFilterOnly(tree), node)
+    )
+
+  let getGroupingResult = async (type) => {
+    let groups = _.get(type, node)
+    if (_.isEmpty(groups)) return {}
+    let hasFieldValuesLast = _.last(groups).type === 'fieldValues'
+
+    let cardinalityNode = await run({
+      ...node,
+      columns: [],
+      rows: [],
+      [type]: groups, // getting only groups for this type
+      expanded: {
+        [type]: true,
+        skipValues: true,
+        cardinality: hasFieldValuesLast || !!exportAllPages,
+      },
+    })
+
+    return _.get('context.results', cardinalityNode)
+  }
+
+  let getGroupingSize = (groupingType, cardinalityResult) => {
+    let getNested = _.get(groupingType)
+    let groupingCardinality = 1 // starting with 1 for the total column/row
+
+    F.walk(getNested)((groupResult, index, parents) => {
+      let groupCardinality = _.getOr(
+        _.size(getNested(groupResult)),
+        `${groupingType}Cardinality`,
+        groupResult
+      )
+      if (!groupCardinality) return
+
+      let grouping = _.get([groupingType, _.size(parents)], node)
+      if (!exportAllPages && grouping.type === 'fieldValues') {
+        groupCardinality = _.min([
+          groupCardinality,
+          _.getOr(10, 'size', grouping),
+        ])
```

Comment:
Eventually, this will also move out since we'll want paging info done in type definitions on the client (e.g. you can check a method like `hasMore` in the future)